_transcripts/058-rust-1.58-1.59.md

```diff
@@ -0,0 +1,1176 @@
+compatibility warnings happens, then you aren't seeing it. So like, you might
+not notice that hey, like if you upgrade your compiler, this thing that removes
+a thing, it might break. And so in practice this means that almost all future
+incompatibility warnings have never been promoted to errors. The Rust compiler
+has lots of these warnings, and so the fact that they're warnings means that
+like, you know, people who compile their crates will see them and say, oh, this
+is probably a problem, I'll upgrade this.
+
+But the Rust team has been very reluctant to ever actually promote any of these
+warnings to actual errors, and thereby remove all the bad code, and all the
+compatibility hacks that are needed for this. Even across editions, too. And so
+this feature is really important because it means that future versions and
+editions of Rust might actually have the gumption to start removing some of
+these obviously broken things, that exist only for compatibility. And that
+shouldn't, you know, exist in any kind of ideal version of Rust. So I think the
+next Rust edition is going to be interesting, because it'll probably— I assume,
+I'm not, you know, I'm not the Rust dictator, but I assume that there will be
+many warnings elevated to errors in the next edition that are of this nature.
+
+__Jon__: Yeah. And I think this is— it's a great way, as you say, to sort of
+unblock work on fixing things that might break some users, but giving people
+enough warning ahead of time, and making sure that they will actually see those
+warnings, you know, if they are attentive to their projects. Even if their
+dependencies are not attentive to new warnings.
+
+There's also the associated `cargo report` command, that gives you, like, a full
+report of the code that will be rejected as well. Which is a really nice way to
+get a sort of complete picture of all of the the potential issues you might run
+into the future. And the hope of course then, is that when a user discovers that
+one of their dependencies has this problem, you can go back and tell the authors
+of the dependency, hey, look, you now have a new warning, even though you
+weren't actively looking for this, here's a fix for it. Or at least now you are
+aware, and hopefully we can drive the ecosystem forward in that way.
+
+There's another neat addition in 1.59 that hits probably a small number of
+people, but it's going to make them very happy, which is the ability to create
+stripped binaries. So in general, Rust tends to develop, or build binaries that
+include debug info. And in particular, it includes debug info from things like
+the standard library. It includes debug info from your own code. There are ways
+to turn it off of course, and it's been possible to strip that out after the
+fact manually. Like you can use the `strip` command, for example. But now Cargo
+and `rustc` support stripping out this debug info when the binary is linked. You
```

Comment:
Is "debug info" right? (debug-info, debuginfo?)

Reply:
https://blog.rust-lang.org/2022/02/24/Rust-1.59.0.html#creating-stripped-binaries uses "debuginfo", so let's use that here too.